### ISSUE TITLE
Custom decimal field validation bug

### ISSUE THREAD

**Issue by blueberry-bytes** [moneymanagerex/moneymanagerex#5999]
MMEX version:
 - [ ] 1.6.0
 - [ ] 1.6.1
 - [ ] 1.6.2
 - [x] 1.6.3 64-bit
 - [ ] Other (please specify)

Operating System:
 - [x] Windows
 - [ ] Mac OSX
 - [ ] Linux 

# Description of the bug
I have a custom decimal field with 2 decimal points.
![image](https://github.com/moneymanagerex/moneymanagerex/assets/5322956/34989347-51b8-4d20-a0a0-9a53d0342df4)
RegEx: [0-9]+,[0-9]{2}

When I write any integer in that field it automatically adds separator and two zeroes. But it won't validate properly because it checks whatever I wrote, not a current field value.
![image](https://github.com/moneymanagerex/moneymanagerex/assets/5322956/042fdd95-ef3c-44d9-bc7b-0947abf07fa5)

# Reproduction

Is the bug reproducible? 
 - [x] Always 
 - [ ] Randomly 
 - [ ] Happened only once

Reproduction steps:
Create custom field as described. Add transaction (or edit existing), write any integer in that custom field.

Expected result:
Validate properly.

Actual result:
Wrong validation. I have to manually add separator and zeroes to get through validation. Nothing else worked for me.

# Additional information

If you have a backtrace for a crash or a warning, paste it here.

**Comment by tactilis** [comment]
There is no need for RegEx validation.

The video below is from 1.6.4 Beta 7 although I don't think there has been any change from 1.6.3.

I type:
12 and press Return
1 and press Return
12,34 and press Return
Delete ,34 and press Return

https://github.com/moneymanagerex/moneymanagerex/assets/4060092/2973a840-28e8-4690-9ce0-f9f647f3363c

I have these settings:

![image](https://github.com/moneymanagerex/moneymanagerex/assets/4060092/4e7220d8-8574-4d64-8e80-5184d101abe2)


![image](https://github.com/moneymanagerex/moneymanagerex/assets/4060092/eaa98a5b-ceab-47a4-8025-915fd9d2e091)

**Comment by blueberry-bytes** [comment]
Ok, removing RegEx helps. I don't remember now, if I put it there or it appeared by default, it was long time ago. But I guess it's still a bug, just a very low priority.

**Comment by tactilis** [comment]
> But I guess it's still a bug

Agreed.

What is happening is that when OK is clicked, the RegEx is being checked against the characters the user typed (in this case 12), instead of the characters in the field (12,00) after the user has pressed Return, has tabbed out of the field, or changed focus as a consequence of clicking OK.

**Comment by n-stein** [comment]
> What is happening is that when OK is clicked, the RegEx is being checked against the characters the user typed (in this case 12), instead of the characters in the field (12,00) after the user has pressed Return, has tabbed out of the field, or changed focus as a consequence of clicking OK.

Correct. The validation logic was checking regex before reformatting the integer/decimal fields. I have simply reordered them to put the regex check at the end.
